docs/design/vm_data_protection-design.md

```diff
@@ -0,0 +1,502 @@
+
+
+#### Purpose
+
+- Proxy access to the Kopia repository.
+- Manage user authentication and authorization.
+- Handle repository unlocking using repository credentials.
+- Provide TLS-secured communication.
+
+#### Spec Fields
+
+| Field                        | Type        | Description                                                                                         |
+|------------------------------|-------------|-------------------------------------------------------------------------------------------------|
+| `backupRepositoryRef`             | string (optional) | Reference to the Velero **BackupRepository (BR)** resource. Cannot be used together with `backupStorageLocationRepositoryRef`.   |
+| `backupStorageLocationRepositoryRef` | string (optional) | Reference to the **BackupStorageLocationRepository (BSLR)** resource. Cannot be used together with `backupRepositoryRef`.|
+| `userSecretRef.name`              | string (optional) | Name of a Secret containing Kopia proxy server user credentials (username/password).             |
+| `loadAffinityConfig`              | array of `LoadAffinity` (optional) | Specifies node affinity rules for the Kopia server pods.   |
+| `podResources`                   | `PodResources` (optional) | Specifies CPU/memory resource requests and limits for pods launched by the BSLS. |
+---
+
+#### Status Fields
+
+| Field              | Type    | Description                                  |
+|--------------------|---------|----------------------------------------------|
+| `phase`            | string  | Current lifecycle phase, e.g., `Ready`.      |
+| `endpoint`         | string  | Network address to reach the Kopia Server.  |
+| `tlsFingerprint`   | string  | SHA-256 fingerprint of the server TLS cert. |
+| `conditions`       | array   | List of current status conditions/events.   |
+
+---
+
+```yaml
+# Example BSLS CRD specification
+apiVersion: oadp.openshift.io/v1alpha1
+kind: BackupStorageLocationServer
+metadata:
+  name: vm-backup-server
+  namespace: openshift-adp
+spec:
+  # Reference to Velero BR (mutually exclusive with backupStorageLocationRepositoryRef)
+  backupRepositoryRef: my-backup-repo
+  # Reference to BSLR (mutually exclusive with backupRepositoryRef)
+  backupStorageLocationRepositoryRef: my-bslr
+  userSecretRef:                             # kopia server user credentials
+    name: kopia-proxy-users
+  loadAffinityConfig:                        # optional node affinity rules
+    - nodeSelector:
+        matchLabels:
+          node-role.openshift.io/bsls: "true"
+  podResources:                             # optional CPU/memory requests and limits
+    requests:
+      cpu: "500m"
+      memory: "1Gi"
+    limits:
+      cpu: "1"
+      memory: "2Gi"
+status:
+  phase: Ready
+  endpoint: bsls-vm-backup-server.openshift-adp.svc.cluster.local:51515
+  tlsFingerprint: sha256:abcd1234ef5678...
+  conditions: []
+```
+
```

Comment:
Is this auth flow correct ? 
VM User → [username/password] → Backup Server → [repo password] → Repository
          ↓
      [TLS fingerprint verification]